### ISSUE TITLE
fn_call_style = "Block" working badly with tab_spaces < 4

### ISSUE THREAD

**Issue by valff** [rust-lang-nursery/rustfmt#1427]
Having following options in `rustfmt.toml`:

```toml
    fn_call_style = "Block"
    max_width = 80
    # tab_spaces = 2
```

This snippet looks beautiful:

```rust
    fn main() {
        exceptions::config(
            move || {
                (NmiConfig {}, HardFaultConfig {}, SysTickConfig { gpio_cbsrr })
            },
        );
    }
```

However when `tab_spaces` option is uncommented, `rustfmt` fails fitting in `max_width`:

```
    Rustfmt failed at src/main.rs:2: line exceeded maximum length (maximum: 80, found: 96) (sorry)
```

Producing this snippet:

```rust
    fn main() {
      exceptions::config(move || (NmiConfig {}, HardFaultConfig {}, SysTickConfig { gpio_cbsrr }),);
    }
```

It is expected to be the same as the previous snippet, except a number of indent spaces.

rustfmt version is 0.8.1

**Comment by topecongiro** [comment]
This is fixed in 0.8.4.
With `tab_spaces = 2`, rustfmt produces the following snippet:
```rust
fn main() {
  exceptions::config(
    move || (NmiConfig {}, HardFaultConfig {}, SysTickConfig { gpio_cbsrr }),
  );
}
```
However, with `tab_spaces = 4`, the output will be:
```rust
fn main() {
    exceptions::config(
        move || {
            (NmiConfig {}, HardFaultConfig {}, SysTickConfig { gpio_cbsrr })
        }
    );
}
```
That is, the trailing comma after the closure is missing.

**Comment by valff** [comment]
Thanks!
